components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -636,16 +731,93 @@ public void updateConsumerApplication(OAuthConsumerAppDTO consumerAppDTO) throws
             oauthappdo.setTokenRevocationWithIDPSessionTerminationEnabled(consumerAppDTO
                     .isTokenRevocationWithIDPSessionTerminationEnabled());
             oauthappdo.setTokenBindingValidationEnabled(consumerAppDTO.isTokenBindingValidationEnabled());
-            oauthappdo.setTokenEndpointAuthMethod(consumerAppDTO.getTokenEndpointAuthMethod());
-            oauthappdo.setTokenEndpointAuthSignatureAlgorithm(
-                    consumerAppDTO.getTokenEndpointAuthSignatureAlgorithm());
+
+            String tokenEndpointAuthMethod = consumerAppDTO.getTokenEndpointAuthMethod();
+            if (StringUtils.isNotEmpty(tokenEndpointAuthMethod)) {
+                if (enforceFAPIDCR) {
+                    validateFAPITokenAuthMethods(tokenEndpointAuthMethod);
+                } else {
+                    filterTokenEndpointAuthMethods(tokenEndpointAuthMethod);
+                }
+                oauthappdo.setTokenEndpointAuthMethod(tokenEndpointAuthMethod);
+            }
+
+            String tokenEndpointAuthSignatureAlgorithm = consumerAppDTO.getTokenEndpointAuthSignatureAlgorithm();
+            if (StringUtils.isNotEmpty(tokenEndpointAuthSignatureAlgorithm)) {
+                if (enforceFAPIDCR) {
+                    validateFAPISignatureAlgorithms(tokenEndpointAuthSignatureAlgorithm);
+                } else {
+                    filterSignatureAlgorithms(tokenEndpointAuthSignatureAlgorithm,
+                            OAuthConstants.TOKEN_EP_SIGNATURE_ALG_CONFIGURATION);
+                }
+                oauthappdo.setTokenEndpointAuthSignatureAlgorithm(tokenEndpointAuthSignatureAlgorithm);
+            }
+
+            if (StringUtils.isNotEmpty(consumerAppDTO.getSubjectType())) {
+                OAuthConstants.SubjectType subjectType = OAuthConstants.SubjectType.fromValue(
+                        consumerAppDTO.getSubjectType());
+                if (subjectType == null) {
+                    consumerAppDTO.setSubjectType(OIDCClaimUtil.getDefaultSubjectType().toString());
+                }
+                if (OAuthConstants.SubjectType.PAIRWISE.getValue().equals(consumerAppDTO.getSubjectType())) {
+                    List<String> callBackURIList = new ArrayList<>();
+                    // Need to split the redirect uris for validating the host names since it is combined
+                    // into one regular expression.
+                    if (consumerAppDTO.getCallbackUrl().startsWith(OAuthConstants.CALLBACK_URL_REGEXP_PREFIX)) {
+                        callBackURIList = getRedirectURIList(consumerAppDTO);
+                        ;
+                    } else {
+                        callBackURIList.add(consumerAppDTO.getCallbackUrl());
+                    }
+                    if (StringUtils.isNotEmpty(consumerAppDTO.getSectorIdentifierURI())) {
+                        validateSectorIdentifierURI(consumerAppDTO.getSectorIdentifierURI(), callBackURIList);
```

Comment:
dont we need to set the SectorIdentifierURI to the app after this?

Reply:
added